sdk/src/main/scala/io/horizen/account/state/ForgerStakeV2MsgProcessor.scala

```diff
@@ -81,21 +87,81 @@ case class ForgerStakeV2MsgProcessor(networkParams: NetworkParams) extends Nativ
     Array.emptyByteArray
   }
 
-}
+  def doActivateCmd(invocation: Invocation, view: BaseAccountStateView, context: ExecutionContext): Array[Byte] = {
+
+    //Check is well formed
+    requireIsNotPayable(invocation)
+    checkInputDoesntContainParams(invocation)
+
+    //Check it cannot called twice
+    if (StakeStorage.isActive(view)) {
+      val msgStr = s"Forger stake V2 already activated"
+      log.debug(msgStr)
+      throw new ExecutionRevertedException(msgStr)
+    }
+
+    val intrinsicGas = invocation.gasPool.getUsedGas
+
+    //Call "disableAndMigrate" on old forger stake msg processor, so it won't be used anymore.
+    //It returns all existing stakes that will be recreated in the ForgerStakes V2
+    val result = context.execute(invocation.call(FORGER_STAKE_SMART_CONTRACT_ADDRESS, BigInteger.ZERO,
+      BytesUtils.fromHexString(ForgerStakeMsgProcessor.DisableAndMigrateCmd), invocation.gasPool.getGas))
+    val listOfExistingStakes = AccountForgingStakeInfoListDecoder.decode(result).listOfStakes
+    val stakesByForger = listOfExistingStakes.groupBy(_.forgerStakeData.forgerPublicKeys)
+
+    val epochNumber = context.blockContext.consensusEpochNumber
+
+    var totalMigratedStakeAmount = BigInteger.ZERO
+    stakesByForger.foreach { case (forgerKeys, stakesByForger) =>
+      // Sum the stakes by delegator
+      val stakesByDelegator = stakesByForger.groupBy(_.forgerStakeData.ownerPublicKey)
+      val listOfTotalStakesByDelegator = stakesByDelegator.mapValues(_.foldLeft(BigInteger.ZERO){
+        (sum, stake) => sum.add(stake.forgerStakeData.stakedAmount)})
+      //Take first delegator for registering the forger
+      val (firstDelegator, firstDelegatorStakeAmount) = listOfTotalStakesByDelegator.head
+      //in this case we don't have to check the 10 ZEN minimum threshold for adding a new forger
+      StakeStorage.addForger(view, forgerKeys.blockSignPublicKey,
+        forgerKeys.vrfPublicKey, 0, Address.ZERO, epochNumber, firstDelegator.address(), firstDelegatorStakeAmount)
+      totalMigratedStakeAmount = totalMigratedStakeAmount.add(firstDelegatorStakeAmount)
+      listOfTotalStakesByDelegator.tail.foreach { case (delegator, delegatorStakeAmount) =>
+        StakeStorage.addStake(view, forgerKeys.blockSignPublicKey, forgerKeys.vrfPublicKey,
+          epochNumber, delegator.address(), delegatorStakeAmount)
+        totalMigratedStakeAmount = totalMigratedStakeAmount.add(delegatorStakeAmount)
+      }
+    }
+
+    //Update the balance of both forger stake msg processors
+    view.subBalance(FORGER_STAKE_SMART_CONTRACT_ADDRESS, totalMigratedStakeAmount)
+    view.addBalance(FORGER_STAKE_V2_SMART_CONTRACT_ADDRESS, totalMigratedStakeAmount)
```

Comment:
Maybe we could check if any remaining fund is left on V1 balance? In that case we might decide to send them to the forger pool or burn it or just logging it.

Reply:
I think we can discuss about it, but it is something that should be decided by the product manager.